### PR TITLE
Use y2logstep post_fail_hook in AY installation

### PR DESCRIPTION
**Pull request by rwx788**
We have a lot of useful features implemented in y2logstep to gather
information in case of failed installation, which we also should use in
AY installation.

See [poo#28851](https://progress.opensuse.org/issues/28851).

#### Verification runs
[YaST log investigation for AY](http://g226.suse.de/tests/2039#downloads)
[extra logs for yast ncurses tests](http://g226.suse.de/tests/2044#downloads)

### CODE REVIEW

File: lib/y2logsstep.pm [hunk 1]
```diff
@@ -238,27 +239,8 @@ sub save_upload_y2logs {
...
-    $self->SUPER::post_fail_hook;
-    get_to_console;
-    $self->get_ip_address;
-    $self->remount_tmp_if_ro;
-    # Avoid collectin logs twice when investigate_yast2_failure() is inteded to hard-fail
-    $self->save_upload_y2logs unless get_var('ASSERT_Y2LOGS');
-    return if is_caasp;
-
+sub save_system_logs {
+    my $self = shift;
```
Comment (SergioAtSUSE) [inline lib/y2logsstep.pm]: Why not using here the recommended?

```perl
my ($self) = @_;
```

File: tests/autoyast/installation.pm [hunk 2]
```diff
@@ -52,10 +52,8 @@ sub save_and_upload_yastlogs {
     my $name = $stage . ($suffix // '');
```
Comment (SergioAtSUSE) [inline tests/autoyast/installation.pm]: I think the function `save_and_upload_yastlogs` needs another name. It is not doing that anymore and it is confusing.

Comment (rwx788) [reply tests/autoyast/installation.pm]: I've missed something here, so thanks a lot for noticing this

**Comment by rwx788** [comment]
Verification runs updated

**Comment by SergioAtSUSE** [comment]
Interesting, travis is complaining, but how is it possible that someone could introduce this extra white space?

**Comment by okurz** [comment]
Let's try to fix tests first. This was introduced by a new version of os-autoinst changing the test API as well as updating the perltidyrc. https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/5355#issuecomment-405826281 fixes a part of it, I am on the second part already

**Comment by okurz** [comment]
Let's try to fix tests first. This was introduced by a new version of os-autoinst changing the test API as well as updating the perltidyrc. https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/5355#issuecomment-405826281 fixes a part of it, I am on the second part already

**Comment by okurz** [comment]
please rebase

**Pull request merged**